### PR TITLE
ci: Enable cache for install dependencies

### PR DESCRIPTION
**Pull request by Yakutoc**
### Мотивация

Необходимо ускорить наш CI. 

Один из способов это, начинать использовать **cache** для ускорения установки зависимостей.

#### Область видимости для cache

> Access restrictions provide cache isolation and security by creating a logical boundary between different branches or tags. > Workflow runs can restore caches created in either the current branch or the default branch (usually main). 

> If a workflow run is triggered for a pull request, it can also restore caches created in the base branch, including base branches of forked repositories.

#### Документация

[actions/setup-node](
https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#working-with-lockfiles)

[actions/cache ](https://github.com/actions/cache)

[restrictions-for-accessing-a-cache](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache)

На скриншотах данные по запускам. Сверху вниз. 

Первый запуск без использования **cache**, последующие запуски уже с ним.       

#### Component Performance Testing

Screenshot 2023-05-30 at 22 51 59

#### Typescript coverage
Screenshot 2023-06-01 at 13 29 00

### Примечания 

На скриншоте для `Typescript coverage` можно увидеть **"аномалию"** с включенным **cache** время исполнения процесса даже дольше (12м39сек), чем у первого запуска. 

В данном случае самый долгий процесс это `npx lerna bootstrap` и он не всегда проходит стабильно за одно и тоже время, бывают отклонения, что и произошло в этом запуске.

Все последующие улучшения связанные с CI будут направлены на стабилизацию и ускорения шага с `npx lerna bootstrap`.                     

-----------

#### Using concurrency

Так же добавили логику для отключения workflows в PR на событие push, что не отключать руками и не засорять ресурсы.        

[Concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency)

---------

:baby_chick: Download canary assets:

[color_sberHealth_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.531.5145055056.swift)  
[color_sberHealth_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.531.5145055056.kt)  
[color_sberHealth_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.531.5145055056.ts)  
[color_sbermarket_business_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.531.5145055056.swift)  
[color_sbermarket_business_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.531.5145055056.kt)  
[color_sbermarket_business_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.531.5145055056.ts)  
[color_sbermarket_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.531.5145055056.swift)  
[color_sbermarket_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.531.5145055056.kt)  
[color_sbermarket_metro_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.531.5145055056.swift)  
[color_sbermarket_metro_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.531.5145055056.kt)  
[color_sbermarket_metro_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.531.5145055056.ts)  
[color_sbermarket_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.531.5145055056.ts)  
[color_sbermarket_selgros_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.531.5145055056.swift)  
[color_sbermarket_selgros_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.531.5145055056.kt)  
[color_sbermarket_selgros_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.531.5145055056.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.531.5145055056.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.531.5145055056.kt)  
[color_sbermarket_wlbusiness_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.531.5145055056.ts)  
[color_sberonline_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.531.5145055056.swift)  
[color_sberonline_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.531.5145055056.kt)  
[color_sberonline_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.531.5145055056.ts)  
[color_sberprime_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.531.5145055056.swift)  
[color_sberprime_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.531.5145055056.kt)  
[color_sberprime_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.531.5145055056.ts)  
[shadow_sbermarket_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.531.5145055056.ts)  
[typo_mage_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.531.5145055056.swift)  
[typo_mage_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.531.5145055056.kt)  
[typo_mage_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.531.5145055056.ts)  
[typo_plasma_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.531.5145055056.swift)  
[typo_plasma_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.531.5145055056.kt)  
[typo_plasma_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.531.5145055056.ts)  
[typo_ruler_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.531.5145055056.swift)  
[typo_ruler_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.531.5145055056.kt)  
[typo_ruler_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.531.5145055056.ts)  
[typo_sage_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.531.5145055056.swift)  
[typo_sage_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.531.5145055056.kt)  
[typo_sage_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.531.5145055056.ts)  
[typo_sbermarket_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.531.5145055056.swift)  
[typo_sbermarket_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.531.5145055056.kt)  
[typo_sbermarket_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.531.5145055056.ts)  
[typo_soulmate_ios-swift--canary.531.5145055056.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.531.5145055056.swift)  
[typo_soulmate_kotlin--canary.531.5145055056.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.531.5145055056.kt)  
[typo_soulmate_react-native--canary.531.5145055056.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.531.5145055056.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.215.1-canary.531.5145055056.0
  npm install @salutejs/plasma-core@1.120.1-canary.531.5145055056.0
  npm install @salutejs/plasma-hope@1.215.1-canary.531.5145055056.0
  npm install @salutejs/plasma-icons@1.149.1-canary.531.5145055056.0
  npm install @salutejs/plasma-temple@1.164.1-canary.531.5145055056.0
  npm install @salutejs/plasma-tokens@1.54.1-canary.531.5145055056.0
  npm install @salutejs/plasma-typo@0.33.1-canary.531.5145055056.0
  npm install @salutejs/plasma-ui@1.196.1-canary.531.5145055056.0
  npm install @salutejs/plasma-web@1.215.1-canary.531.5145055056.0
  npm install @salutejs/plasma-cy-utils@0.61.1-canary.531.5145055056.0
  npm install @salutejs/plasma-sb-utils@0.118.1-canary.531.5145055056.0
  # or 
  yarn add @salutejs/plasma-b2c@1.215.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-core@1.120.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-hope@1.215.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-icons@1.149.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-temple@1.164.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-tokens@1.54.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-typo@0.33.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-ui@1.196.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-web@1.215.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-cy-utils@0.61.1-canary.531.5145055056.0
  yarn add @salutejs/plasma-sb-utils@0.118.1-canary.531.5145055056.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-531/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-531/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-531/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-531/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-531/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-531/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-531/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-531/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-531/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-531/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-531/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-531/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-531/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-531/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-531/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-531/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-531/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-531/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-531/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-531/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-531/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-531/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-531/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by akhdrv — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-531/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-531/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-531/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-531/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-531/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-531/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-531/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yeti-or — APPROVED** [review]

**Review by kayman233 — APPROVED** [review]

**Review by kayman233 — APPROVED** [review]

File: .github/workflows/cypress-ct-17.yml [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
-  push:
+  pull_request:
+    branches:
+      - master
+      - dev
+
+concurrency:
+  # New commit on branch cancels running workflows of the same branch
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
```
Comment (neretin-trike) [inline .github/workflows/cypress-ct-17.yml:13]: а тут точно необходимо `concurrency`? Если я правильно понял смысл, то от отменяет все запущенные джобы, если произошёл новый пуш. У нас такого вроде не должно быть, т.к. мы merge queue исползьуем?

File: .github/workflows/cypress-ct.yml [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
-  push:
+  pull_request:
+    branches:
+      - master
+      - dev
+
+concurrency:
+  # New commit on branch cancels running workflows of the same branch
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
```
Comment (neretin-trike) [inline .github/workflows/cypress-ct.yml:13]: тот же вопросик

File: .github/workflows/performance-test.yml [hunk 3]
```diff
@@ -6,6 +6,11 @@ on:
       - master
       - dev
 
+concurrency:
+  # New commit on branch cancels running workflows of the same branch
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
```
Comment (neretin-trike) [inline .github/workflows/performance-test.yml:13]: и тут

File: .github/workflows/cypress-ct-17.yml (hunk 1 above)
Comment (Yeti-or) [reply .github/workflows/cypress-ct-17.yml:13]: я понял что это на случай если ты допушил слишком быстро

Comment (Yakutoc) [reply .github/workflows/cypress-ct-17.yml:13]: @neretin-trike 

`group: ${{ github.workflow }}-${{ github.head_ref }}` как раз формирует область применения отмены.

Comment (neretin-trike) [reply .github/workflows/cypress-ct-17.yml:13]: ага, Саша объяснил 👌

**Review by neretin-trike — APPROVED** [review]

File: .github/workflows/cypress-ct-17.yml (hunk 1 above)
Comment (Yakutoc) [reply .github/workflows/cypress-ct-17.yml:13]: @neretin-trike 

`group: ${{ github.workflow }}-${{ github.head_ref }}` как раз формирует область применения отмены.

**Review by neretin-trike — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-531/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-531/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-531/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-531/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-531/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-531/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**